### PR TITLE
SG-3 - Demonstrating UpgradeSchema and UpgradeData scripts by adding a gender column and updating the example record

### PR DESCRIPTION
**Pull request by msherer**
Adding a gender column to the mediotype_employee table and updating our example Joel Hart record in specific with a gender value.

Note: The reason behind not getting a collection and looping through is because we have 1 explicit example record and any data added between versions 0.0.1 and 0.0.2 cannot be assumed to be a single gender.

### CODE REVIEW

File: app/code/Mediotype/Employee/Setup/UpgradeData.php [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+     */
+    public function upgrade(ModuleDataSetupInterface $setup, ModuleContextInterface $context)
+    {
+        $setup->startSetup();
+
+        $tableName = 'mediotype_employee';
+
+        /**
+         * Here we check if our module is of version 0.0.2 then update our Joel Hart record. If the version
+         * of our module is less than 0.0.2 version_compare will return -1 and we will know that we need to
```
Comment (LeeSaferite) [inline app/code/Mediotype/Employee/Setup/UpgradeData.php]: In this case `version_compare` returns a boolean value, not an integer.

File: app/code/Mediotype/Employee/Setup/UpgradeData.php [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+     *
+     * @param \Magento\Framework\Setup\ModuleDataSetupInterface $setup
+     * @param string $tableName
+     */
+    private function updateGender($setup, $tableName)
+    {
+        /**
+         * Check if our row exists in the database.
+         */
+        if ($setup->getTableRow($setup->getTable($tableName), 'employee_id', 1)) {
```
Comment (LeeSaferite) [inline app/code/Mediotype/Employee/Setup/UpgradeData.php]: Why wouldn't you upgrade every record? If data was added between deploying version 0.0.1 and 0.0.2 then those records will not have gender data yet, right?

Comment (msherer) [reply app/code/Mediotype/Employee/Setup/UpgradeData.php]: We have no way of determining what gender every record would be considered. The only constant during the upgrade is that our example record is Joel and we know he is a male. Every other record should be defaulted to NULL and then we can update them on a record by record basis.

File: app/code/Mediotype/Employee/Setup/UpgradeData.php (hunk 1 above)
Comment (msherer) [reply app/code/Mediotype/Employee/Setup/UpgradeData.php]: Ah good catch, I was looking at the result of not passing in an operator in the function call similar to our discussion earlier.

**Comment by jkenneydaniel** [comment]
Not sure if it should be separate (or considered part of the scope for this PR) - but you're adding a Gender column and not updating the interface that is supposed to represent the data.

**Comment by msherer** [comment]
@jkenneydaniel, @magedevjosh had me add getters and setters to SG-4 in the interface, so we don't need to wait for them here. Unless you have any other concerns or changes I would appreciate one last review.

**Pull request closed without merging**